src/Majora/Framework/Model/LazyPropertiesTrait.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Majora\Framework\Model;
+
+/**
+ * Trait which provides lazy load delegates call on fields.
+ */
+trait LazyPropertiesTrait
+{
+    /**
+     * @var EntityCollection
+     */
+    private $delegatesCollection;
+
+    /**
+     * @see LazyPropertiesInterface::registerLoader()
+     */
+    public function registerLoader($field, \Closure $delegate)
```

Comment:
Yes because "callable" have to be invoked by ```call_user_func()``` which allow some weird architecture behavior like call another class, thats something i dont want.